thirdparty/ipfsaddr/ipfsaddr.go

```diff
@@ -106,6 +107,13 @@ func ParseMultiaddr(m ma.Multiaddr) (a IPFSAddr, err error) {
 
 func Transport(iaddr IPFSAddr) (maddr ma.Multiaddr) {
 	maddr = iaddr.Multiaddr()
+
+	// /ipfs/QmId is part of the transport address for p2p-circuit
+	_, err := maddr.ValueForProtocol(circuit.P_CIRCUIT)
```

Comment:
Added TODO comment in https://github.com/ipfs/go-ipfs/pull/4091/commits/a11f632244df268fbbbdbbb1cec6c2e2a50e7708